src/main/java/io/openliberty/tools/common/plugins/util/DevUtil.java

```diff
@@ -2698,6 +2703,7 @@ public void watchFiles(File outputDirectory, File testOutputDirectory, final Thr
             // check for upstream projects
             if (isMultiModuleProject()) {
                 for (ProjectModule p : upstreamProjects) {
+                    // TODO: Consider whether or not we should generate features at this point
```

Comment:
Why would we generate features here?

Reply:
I am thinking that we probably do not want generate features at this point, but the way the code is currently written, we may re-generate features inside updateArtifacts if compilation dependencies have changed. I am not sure if that is the behaviour we want at start up or not, which is why I added the TODO.

Reply:
Thanks, that makes sense! I agree we do not want to generate features. I think it is safe to leave that logic in updateArtifacts for now in the off chance that the dependencies have changed since dev mode startup.